### PR TITLE
Add desc of new TiFlash storage settings and guidelines for upgrading in v4.0.9

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What is changed, added or deleted? (Required)

In v4.0.9, TiFlash supports new configurations for better performance when it is deployed on multi-disk node.

This PR
* Add some description about the new configurations.
* Add guidelines for how to enable those new configurations after upgrading the TiDB cluster.

Please do NOT merge this PR until v4.0.9 is ready for release~

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: N/A
- Other reference link(s): https://github.com/pingcap/tiup/pull/931
- Related English version PR: https://github.com/pingcap/docs/pull/4290

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@JaySon-Huang Please involve a technical review~

**Comment by JaySon-Huang** [comment]
> @JaySon-Huang Please involve a technical review~

@flowbehappy

File: config-templates/complex-tiflash.yaml
Comment (JaySon-Huang) [inline config-templates/complex-tiflash.yaml:46]: Related PR: https://github.com/pingcap/tics/issues/1223

File: config-templates/complex-tiflash.yaml
Comment (flowbehappy) [inline config-templates/complex-tiflash.yaml:107]:
```suggestion
    ## The path to store data. Using ',' to separate the paths if you need multiple paths.
    ## Note that starting from v4.0.9, `data_dir` is deprecated, in favor of the new `config.storage.*` configurations. 
    ## Thought `data_dir` remains supported, it is recommended to use the new configuration format.
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:33]:
```suggestion
# tiflash TCP/HTTP 等辅助服务的监听 host。建议配置成 0.0.0.0，即监听本机所有 IP 地址。
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:78]:
```suggestion
    ## 用于存储最新的数据。该目录需要实时写入从 TiKV 同步过来的更新，故需要较高的 IOPS。
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:65]:
```suggestion
    ## 用于存储主要的数据。该目录数据占总数据的 90% 以上。
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:85]:
```suggestion
    ## 用于存储 Raft Log 数据，数据量很小。要实时写入从 TiKV 同步过来的数据，需要较高的 IOPS。
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:87]:
```suggestion
```

I don't think we should expose this configuration to users. They are 99.9% use the same path as `storage.latest.dir`.

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:167]:
```suggestion
> [storage] 参数在 TiUP v1.2.5 版本开始支持。如果你的 TiDB 版本为 v4.0.9 及以上，请确保你的 TiUP 版本不低于 v1.2.5，否则 [storage] 中定义的数据目录不会被纳入 TiUP 管理。
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:157]:
```suggestion

- **如果你部署的 TiDB 集群版本 >= v4.0.9：**

```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:169]:
```suggestion
- **如果你部署的 TiDB 集群版本 < v4.0.9：**

```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:122]:
```suggestion
```
I suggest to remove it.

Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:122]:
```suggestion
```
I suggest to remove it.

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:179]:
```suggestion
#### 升级节点版本至 v4.0.9 及以上
```

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:181]:
```suggestion
v4.0.9 之前的版本中，TiFlash 只支持将存储引擎中的主要数据分布在多盘上。v4.0.9 及之后的版本中，TiFlash 支持将存储引擎的主要数据、新数据都分布在多盘上。
```

File: tiflash/tiflash-configuration.md
Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:140]: I don't think it is a good idea to define a list following with multiple lines

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md]: Use subtitle instead of list.

File: tiflash/tiflash-configuration.md
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:52]: Can we show the default path here?

**Review by flowbehappy — APPROVED** [review]
LGTM with a minor comment.

**Comment by JaySon-Huang** [comment]
Technical review is done. Please help us revise and proofread the English version.
Note that don't merge this PR until v4.0.9 is released.

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:138]:
```suggestion
TiFlash 支持单节点多盘部署。如果你的部署节点上有多块硬盘，可以通过以下的方式配置参数，提高节点的硬盘 I/O 利用率。TiUP 中参数配置格式参照[详细 TiFlash 配置模版](https://github.com/pingcap/docs-cn/blob/master/config-templates/complex-tiflash.yaml)。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:140]:
```suggestion
#### TiDB 集群版本为 v4.0.9 及以上
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:153]: 把旧方法放在新方法前面，似乎更符合逻辑。
```suggestion
#### TiDB 集群版本低于 v4.0.9
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:142]: "低版本的参数仍然支持"是说 v4.0.9 以后仍然支持使用旧方法部署吗？
```suggestion
TiDB v4.0.9 及之后的版本中，TiFlash 支持将存储引擎的主要数据和新数据都分布在多盘上。多盘部署时，推荐使用 `[storage]` 中的参数，以更好地利用节点的 I/O 性能。但 TiFlash 仍然支持 [TiDB 集群版本低于 v4.0.9](#tidb-集群版本低于-v409) 中的参数。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:144]:
```suggestion
如果节点上有多块相同规格的硬盘，推荐把硬盘目录填到 `storage.main.dir` 列表中，`storage.latest.dir` 列表留空。TiFlash 会在所有存储目录之间分摊 I/O 压力以及进行数据均衡。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:146]:
```suggestion
如果节点上有多块规格不同的硬盘，推荐把 I/O 性能较好的硬盘目录配置在 `storage.latest.dir` 中，把 I/O 性能较一般的硬盘目录配置在 `storage.main.dir` 中。例如节点上有一块 NVMe-SSD 硬盘加上两块 SATA-SSD 硬盘，你可以把 `storage.latest.dir` 设为 `["/nvme_ssd_a/data/tiflash"]` 以及把 `storage.main.dir` 设为 `["/sata_ssd_b/data/tiflash", "/sata_ssd_c/data/tiflash"]`。TiFlash 会根据两个目录列表分别进行 I/O 压力分摊及数据均衡。需要注意此情况下，`storage.latest.dir` 中规划的容量大小需要占总规划容量的约 10%。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:150]:
```suggestion
> * `[storage]` 参数从 TiUP v1.2.5 版本开始支持。如果你的 TiDB 版本为 v4.0.9 及以上，请确保你的 TiUP 版本不低于 v1.2.5，否则 `[storage]` 中定义的数据目录不会被 TiUP 纳入管理。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:151]:
```suggestion
> * 在 TiFlash 节点改为使用 `[storage]` 配置后，如果将集群版本降级到低于 v4.0.9，可能导致 TiFlash 部分数据丢失。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:148]: 涉及丢数据的内容，需要提高级别。
```suggestion
> **警告：**
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:155]:
```suggestion
TiDB v4.0.9 之前的版本中，TiFlash 只支持将存储引擎中的主要数据分布在多盘上。通过 `path`（TiUP 中为 `data_dir`）和 `path_realtime_mode` 这两个参数配置多盘部署。
```

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:159]:
```suggestion
如果节点上有多块相同规格的硬盘，可以把 `path_realtime_mode` 参数留空（或者把该值明确地设为 `false`）。这表示数据会在所有的存储目录之间进行均衡。但由于最新的数据仍然只会被写入到第一个目录，因此该目录所在的硬盘会较其他硬盘繁忙。
```

File: tiflash/tiflash-configuration.md
Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:64]: `*` 的三行是描述 storage.main.capacity 值的三个约束

File: tiflash/tiflash-configuration.md
Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:142]: 是的

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [reply tiflash/tiflash-configuration.md:64]: 所以星号表示列表吗？

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:35]: 配置项和其他配置项的注释之间可以空一行吗

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:35]: 目前一些简单的配置项之间没有加空行。而是按照功能上，在相近的一些参数之间没有加空行。
担心如果全都加上空行，会不会视觉上觉得整个文件很冗长……

**Comment by TomShawn** [comment]
LGTM

File: tiflash/tiflash-configuration.md
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:42]:
```suggestion
minmax_index_cache_size = 5368709120
## DeltaIndex 内存 cache 大小限制，默认为 0，代表没有限制
delta_index_cache_size = 0
```

**Comment by TomShawn** [comment]
@JaySon-Huang Could you please resolve the conflict before this PR is merged? thanks!

**Review by TomShawn — APPROVED** [review]

**Pull request merged**
